books.js

```diff
@@ -1,18 +1,137 @@
-const bookList = [];
-
-class Books {
-  constructor(title, author) {
+/* eslint-disable max-classes-per-file */
+/* eslint-disable no-unused-vars */
+class Book2 {
+  constructor(title, author, next_node = null) {
     this.title = title;
     this.author = author;
   }
 }
 
+class BookList2 {
+  constructor() {
+    this.head = null;
+    this.tail = null;
+  }
+
+  add(title, author) {
+    const newnode = new Book2(title, author);
+    if (this.tail === null) {
+      this.head = newnode;
+      this.tail = newnode;
+    } else {
+      this.tail.next_node = newnode;
+      this.tail = newnode;
+    }
+  }
+
+  size() {
+    let count = 0;
+    let node = this.head;
+    while (node) {
+      count += 1;
+      node = node.next_node;
+    }
+    return count;
+  }
+
+  print() {
+    let nodes = this.head;
+    while (nodes) {
+      nodes = nodes.next_node;
+      return `${nodes.title} ${nodes.author}`;
+    }
+    return null;
+  }
+
+  printArray() {
+    const arr = [];
+    let node = this.head;
+    while (node) {
+      const obj = {
+        title: node.title,
+        author: node.author,
+      };
+      arr.push(obj);
```

Comment:
- [ ] OPTIONAL
It will be a good practice to favor the use of array methods such as `concat` over `push`, which doesn't mutate the array.